src/discordas.ts

```diff
@@ -13,223 +13,187 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-    Request,
-    thirdPartyLookup,
-    unstable as Unstable,
-} from "matrix-appservice-bridge";
+import { Appservice, IAppserviceRegistration, LogService } from "matrix-bot-sdk";
 import * as yaml from "js-yaml";
 import * as fs from "fs";
 import { DiscordBridgeConfig } from "./config";
 import { DiscordBot } from "./bot";
 import { DiscordStore } from "./store";
 import { Log } from "./log";
 import "source-map-support/register";
-import { MetricPeg, PrometheusBridgeMetrics } from "./metrics";
+import * as cliArgs from "command-line-args";
+import * as usage from "command-line-usage";
+import * as uuid from "uuid/v4";
 import { IMatrixEvent } from "./matrixtypes";
-import { wrapError, isInstanceOfTypes } from "./util";
+import { MetricPeg, PrometheusBridgeMetrics } from "./metrics";
 
 const log = new Log("DiscordAS");
 
-const cli = new Cli({
-    bridgeConfig: {
-        affectsRegistration: true,
-        schema: "./config/config.schema.yaml",
-    },
-    generateRegistration,
-    registrationPath: "discord-registration.yaml",
-    run,
-});
-
-try {
-    cli.run();
-} catch (err) {
-    log.error("Failed to start bridge.");
-    log.error(err);
+const commandOptions = [
+    { name: "config", alias: "c", type: String },
+    { name: "url", alias: "u", type: String },
+    { name: "port", alias: "p", type: Number },
+    { name: "file", alias: "f", type: String },
+    { name: "generate-registration", alias: "r", type: Boolean },
+    { name: "help", alias: "h", type: Boolean },
+];
+
+function generateRegistration(opts, registrationPath)  {
+    if (!opts.url) {
+        throw Error("'url' not given in command line opts, cannot generate registration file");
+    }
+    const reg = {
+        as_token: uuid(),
+        hs_token: uuid(),
+        id: "discord-bridge",
+        namespaces: {
+            aliases: [
+                {
+                    exclusive: true,
+                    regex: "#_discord_.*",
+                },
+            ],
+            rooms: [ ],
+            users: [
+                {
+                    exclusive: true,
+                    regex: "@_discord_.*",
+                },
+            ],
+        },
+        protocols: ["discord"],
+        rate_limited: false,
+        sender_localpart: "_discord_bot",
+        url: opts.url,
+    } as IAppserviceRegistration;
+    fs.writeFileSync(registrationPath, yaml.safeDump(reg));
 }
 
-function generateRegistration(reg, callback)  {
-    reg.setId(AppServiceRegistration.generateToken());
-    reg.setHomeserverToken(AppServiceRegistration.generateToken());
-    reg.setAppServiceToken(AppServiceRegistration.generateToken());
-    reg.setSenderLocalpart("_discord_bot");
-    reg.addRegexPattern("users", "@_discord_.*", true);
-    reg.addRegexPattern("aliases", "#_discord_.*", true);
-    reg.setRateLimited(false);
-    reg.setProtocols(["discord"]);
-    callback(reg);
-}
+function setupLogging() {
+    const logMap = new Map<string, Log>();
```

Comment:
```suggestion
    		// log matrix-bot-sdk with our logging
		const logMap = new Map<string, Log>();
		// tslint:disable-next-line:no-any
		const logFunc = (level: string, module: string, args: any[]) => {
			if (!Array.isArray(args)) {
				args = [args];
			}
			// filter out spam logging
			if (args.find((s) => s.includes && s.includes("M_USER_IN_USE"))) {
				return;
			}
			let mod = "bot-ask-" + module;
			const modParts = module.match(/^(\S+)\s(.*)/);
			if (modParts) {
				const MOD_PARTS_NORMAL = 1;
				const MOD_PARTS_EXTRA = 2;
				if (modParts[MOD_PARTS_EXTRA]) {
					args.unshift(modParts[MOD_PARTS_EXTRA]);
				}
				mod = "bot-sdk-" + modParts[MOD_PARTS_NORMAL];
			}
			let logger = logMap.get(mod);
			if (!logger) {
				logger = new Log(mod);
				logMap.set(mod, logger);
			}
			logger[level](...args);
		};
		LogService.setLogger({
			// tslint:disable-next-line:no-any
			debug: (mod: string, args: any[]) => logFunc("debug", mod, args),
			// tslint:disable-next-line:no-any
			info: (mod: string, args: any[]) => logFunc("info", mod, args),
			// tslint:disable-next-line:no-any
			warn: (mod: string, args: any[]) => logFunc("warn", mod, args),
			// tslint:disable-next-line:no-any
			error: (mod: string, args: any[]) => logFunc("error", mod, args),
		});```

Reply:
I don't know if we should suggest whole features in reviews..., it's impossible to know what this is replacing.

Reply:
This is replacing the matrix-bot-sdk logger stuffs you did, it improves upon that slightly (that it uses our logger)